File: Sorting/Binary_search.py
# Binary search using python

def binary_search(a,l,h,key):
    mid = (l+h)//2
    if(l > h):
        return -1
    if(a[mid] == key):
        return mid
    if(a[mid] > key):
        return binary_search(a,l,mid-1,key)
    return binary_search(a,mid-1,h,key)
n = int(input("Enter Size of list:- "))
a = []
for i in range (0,n):
    p = int(input())
    a.append(p)
a.sort();
key = int(input("Enter Key:- "))
index = binary_search(a,0,n-1,key)
if(index == -1):
    print("Data not found")
else:
    print("Data found at:- ",index)

File: Sorting/Binary_search.cpp
#include<bits/stdc++.h>
using namespace std;
int binary_search(int *arr,int l,int h,int num)
{
    if(l > h)
    {
        return -1;
    }
    int mid = (l+h)/2;
    if(arr[mid] == num)
    {
        return mid;
    }
    if(arr[mid] > num)
    {
        return binary_search(arr,l,mid-1,num);
    }
    return binary_search(arr,mid+1,h,num);
}
int main()
{
    int n;
    cout << "Enter size of array:- ";
    cin >> n;
    int arr[n];
    int l=0,h=n-1,num;
    cout << "Enter data:- ";
    for(int i=0;i<n;i++)
    {
        cin >> arr[i];
    }
    cout << "Enter data to search:- ";
    cin >> num;
    sort(arr,arr+n);
    int index = binary_search(arr,l,h,num);
    if(index == -1)
    {
        cout << "Data not found";
    }
    else{
        cout << "Data found at index:- " << index << endl;
    }
    return 0;
}
